modules/example-browser/src/caching_materials_manager.ts

```diff
@@ -0,0 +1,201 @@
+   * Both encrypt and decrypt requests count independently towards this total.
+   * Elements will be actively removed from the cache.
+   * The default frequency is to check one item every minute.
+   * This can be configure by passing a `proactiveFrequency`
+   * as the second paramter to however often you want to check in milliseconds.
+   */
+  const capacity = 100
+  const cache = getLocalCryptographicMaterialsCache(capacity)
+
+  /* The partition name lets multiple caching CMMs share the same local cryptographic cache.
+   * If you want these CMMs to all cache the same items,
+   * make the partition name the same.
+   * If no partition is supplied a random one will be generated.
+   * This is so that sharing elements in the cache MUST be an intentional operation.
+   */
+  const partition = 'local partition name'
+
+  /* maxAge is the time in milliseconds that an entry will be cached.
+   * Elements are actively removed from the cache.
+   */
+  const maxAge = 1000 * 60
+
+  /* The maximum amount of bytes that will be encrypted under a single data key.
+   * This value is optional,
+   * but you should configure the lowest value possible.
+   */
+  const maxBytesEncrypted = 100
+
+  /* The maximum number of messages that will be encrypted under a single data key.
+   * This value is optional,
+   * but you should configure the lowest value possible.
+   */
+  const maxMessagesEncrypted = 10
+
+  const cachingCMM = new WebCryptoCachingMaterialsManager({
+    backingMaterials: keyring,
+    cache,
+    partition,
+    maxAge,
+    maxBytesEncrypted,
+    maxMessagesEncrypted
+  })
+
+  /* Encryption context is a *very* powerful tool for controlling and managing access.
+   * It is ***not*** secret!
+   * Encrypted data is opaque.
+   * You can use an encryption context to assert things about the encrypted data.
+   * Just because you can decrypt something does not mean it is what you expect.
+   * For example, if you are are only expecting data from 'us-west-2',
+   * the origin can identify a malicious actor.
+   * See: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#encryption-context
+   *
+   * DATA KEYS FOR MESSAGES WILL ***ONLY*** BE SHARED FOR EXACT MATCHES OF ENCRYPTION CONTEXT.
+   * See: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/data-caching-details.html#caching-encryption-context
+   */
+  const encryptionContext = {
+    stage: 'demo',
+    purpose: 'simple demonstration app',
+    origin: 'us-west-2'
+  }
+
+  /* Find data to encrypt. */
+  const plainText = new Uint8Array([1, 2, 3, 4, 5])
+
+  /* Encrypt the data.
+   * The caching CMM only reuses data keys
+   * when it know the length (or an estimate) of the plaintext.
+   * However, in the browser,
+   * you must provide all of the plaintext to the encrypt function.
+   * Therefore, the encrypt function in the browser knows the length of the plaintext
+   * and does not accept a plaintextLength option.
+   */
```

Comment:
Ryan, is there a reason to explain why you can't provide a value? This function doesn't have this parameter. I would just omit lines 148-153. (Feel free to overrule me if I've misinterpreted this.)

Reply:
If you provide a value and it does not match, this just makes for confusion.